### PR TITLE
ILIAS 10: Add Contact privacy paper

### PR DESCRIPTION
**Pull request by iszmais**
This PR provides a `PRIVACY.md` file for the `Contact` component.

### CODE REVIEW

**Comment by iszmais** [comment]
@mjansenDatabay 3 Open Points from my side:
* > A contact can be used for 2 actions:

Do you know any more actions regarding the contacts directly?
* > If a user is deleted, [...] His mailing lists will not be removed.

I would consider this more as a bug

* This contact system does not seem to work in the current trunk therefore i based this on the current released state and checked for any changes. Im not sure if this affects the privacy since it does not guarantee the functionality in general like tests etc.

Pls leave me some Feedback

Greetings,
@iszmais

**Comment by iszmais** [comment]
I removed the Bug protocol since this is takled [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7437)

@mjansenDatabay Currently OSD Notifications are not deleted on user deletion since their are not tracable back to the user after creation (similar to mail contents). I Added this information to the privacy. Do you see this as an concern?

Greetings,
@iszmais

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The "Contact Service" of ILIAS provides a channel for users to contact each other.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The component only handles the contact and shares of information via the contact functionality.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `There are multiple other ways of context related contacts in ILIAS that are not part of this component, like course member/tutor relations or test correction processes, etc..

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The global activation of the service can be checked with the`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The personal activation of the service can be toggled with the`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `Furthermore, user profiles can be published to give all users access to request contact via the "Who is online" tool.`

And: This does not depend on the publication of user profiles. It only depends on the personal "Visibility" setting "Allow Contact Requests" if a User A is able to initiate a contact request for User B.

Furthermore, it is currently not possible to initiate a request in the "Who is online?" tool. You only can "Accept" or "Reject" open requests there.

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `A contact can be used for two actions:`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `Furthermore, approved contacts are shown within the "Who is online?" tool`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: The "Contact" service is not aware of the data being passed through mails or chats. It does not know anything about

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `Each contact added as an entry to a mailing list is stored by using the user_id.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing settings) and control elements presenting the user's options for contact requests ("Accept/Reject").`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md:52]: We should use another word for `ignorance`.

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `A list of all contacts is shown as a selection when a user adds members to a newly created mailing list.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `A contact relation cannot be perceived within the user interface by any user other than the contacts themselves.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `If a contact request is ignored, the flagged entry in the database will be preserved to prevent another request.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: We should avoid `his` and `him`.

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The "Contact" service does not provide any export option itself.`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. The notification contains`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `Otherwise the information of contacts is not stored in any export of the related components like "Mail", "Chatroom" or "Object".`

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `The notification is handled by the "Notification" component, is self persitent, and therefore not depending on the`

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Thx Ingmar, I added some inline remarks we should address before a discussion with the "Privacy Experts".

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md]: This should mean that, after a profile is published, the "Who is online" tool provides a link to the profile and therefore also a way to request a contact on the profile page indirectly through a public access point (beside closed access from profiles through course etc).

I will clarify that.

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md]: Changed the options to their actual wording ("Confirm contact"/"Ignore contact")

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md:52]: Changed to "ignoring" but i would still set to the word root since this is also the translation of the UI option.

**Comment by iszmais** [comment]
@mjansenDatabay Thx alot. I corrected all phrasings and also got rid of the outdated "Service" wording generally.

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md:88]: We should not use `ILIASObject` here. If you mean `Services/Object` user `Object` as component name.

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]: `to prevent another request.` -> maybe better?: `to prevent a repeated contact request from the same initiator`

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
In my second review, I noticed that three "parts" are missing:

1. MyCourses/MyGroups presents users assigned to the groups and courses of the acting user
2. If you click the "Mail to Members" button in membership-enabled objects, the following GUI to select local roles or display users of the object is part of the "Contact" component.
3. If you enter the "Mail System" and click on the "Search" button, the GUI to search for users/contacts/groups is part of the "Contact" component.

These three should be contained in the "Data being presented" section.

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md:88]: The component `Object` was renamed to `ILIASObject` in ILIAS 10

**Comment by iszmais** [comment]
Hi @mjansenDatabay ,

i found and adapt the third Point but couldnt find any hint for the first 2:

> 1. MyCourses/MyGroups presents users assigned to the groups and courses of the acting user
> 2. If you click the "Mail to Members" button in membership-enabled objects, the following GUI to select local roles or display users of the object is part of the "Contact" component.

All i found arent part of the Contact itself but only avatar presentations.
Can you elaborate this any further?

Greetings,
@iszmais

**Comment by iszmais** [comment]
@mjansenDatabay i added the cases and fixed the missing thread. Pls have a look.

Greetings,
@iszmais

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]:
```suggestion
- Furthermore, a list of user accounts with membership assignment is shown by clicking the
 "List Members" action for one or multiple objects within the "My Courses/My Groups"
```

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]:
```suggestion
  view, accessible directly from the "Compose" view of "Mail" component, or by clicking the "Contacts"
  link item in the main bar. For each account, the username is presented. Additionally, full names are
  presented according to the "Profile and Privacy > Pubslish Profile > Visible for logged in Users"
  setting. The "Contact Status" is shown as well.
```

File: components/ILIAS/Contact/PRIVACY.md
Comment (mjansenDatabay) [inline components/ILIAS/Contact/PRIVACY.md]:
```suggestion
- The "Mail to Members" view of membership-enabled object types, which is also located in the "Contact"
 component, provides a list of participating user accounts. For each account,
 the username is presented. Additionally, full names are presented according to
 the "Profile and Privacy > Pubslish Profile > Visible for logged in Users" setting.
 The assigned local roles in the current object context are shown as well.
```

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Review by mjansenDatabay — APPROVED** [review]

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [inline components/ILIAS/Contact/PRIVACY.md]: I have no problem with Toast, but I have found that less experienced users get confused by it. I think it might be useful to add a short description or rephrase a sentence, maybe smth like: 
"A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that contains the user presentation name (depending on his visibility) and the user intention of the contact request." 
or (more detailed): 
"A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that is displayed in the top right-hand corner of the screen and contains the user presentation name (depending on his visibility) and the user intention of the contact request."  - But I'm perfectly fine with "Toast is a notification element".

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [inline components/ILIAS/Contact/PRIVACY.md]: I would like to specify "actively ignore". Imo "ignore" suggests that no action is necessary, so "actively ignore" sounds like  "actively doing smth that doesn't request an action". Suggestion: When the requested user actively ignores the request (by choosing the option "Ignore contact")...

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [inline components/ILIAS/Contact/PRIVACY.md:53]: Lines 48-52: does this information belong there or more likely in General information? There is no information about presented data  in these lines.

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [inline components/ILIAS/Contact/PRIVACY.md]: Is it a kind of integrated service? If so, please add a section "Integrated services"/"Services being used" (here are some examples: https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/components/ILIAS/ItemGroup/PRIVACY.md or https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/components/ILIAS/File/PRIVACY.md ) and put this information in this new section. 
If it's not, please ignore this comment.
Some explanation: When I asked you to write privacy.md, the section "Integrated Services" was missing from the"How to write a privacy.md" guide, or did not exist as a separate section. We are in the process of revising the guide and adding this section. We apologize for any inconvenience caused :)

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md:53]: This line is not meant to target the functionality, but the active visible options in the dropdown which are a part of the presentation IMO since they imply certain states of requests.

I tried to clarify this with a more detailed writing. Please let me know what you think of it.

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md]: I choose the short one because I tried to balance the description here. I agree that this should be more explanatory, but at the same time this part should not set properties of the toast and therefore imply a restriction on it. (e.g. "in the top right-hand corner of the screen" is a toast property and should be defined by toast and not the contact, especially since this is not relevant for the contact-concerning part here)

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md]: This is more a "Used by services" than "Services being used".

At the end, the notification is a completely independent persistent layer on which the contact has no effect retroactively.

It is like the invitation mail to the course service.

Comment (kseniialf) [reply components/ILIAS/Contact/PRIVACY.md]: okay, got it. Thank you for explaining!

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [reply components/ILIAS/Contact/PRIVACY.md:53]: Thank you for the detailed description, @iszmais ! 
Just for me to clarify: If an user is responding to a request (by choosing one of these options), then a corresponding data entry will be stored. That's why these options are important, did I understand it correctly? There isn't any personal data visible, but choosing the option triggers the personal data processing, doesn't it?

File: components/ILIAS/Contact/PRIVACY.md
Comment (kseniialf) [inline components/ILIAS/Contact/PRIVACY.md:46]:
```suggestion
- The presentation includes a presentation of the user's profile (at least the username, other information are restricted depending on the user's profile publishing settings)
```
afaik the user name is always displayed, even if a requesting user does not make their profile public.

File: components/ILIAS/Contact/PRIVACY.md
Comment (iszmais) [reply components/ILIAS/Contact/PRIVACY.md:53]: @ksgribchenko exactly!
And further, the possible options imply the stored data.
E.g. If the option "Unlink" is visible, this implies that a contact relation to this user exists.
Therefore, it provides the same information as displaying the actual request state directly.

**Review by kseniialf — COMMENTED** [review]
Thank you for accepting the changes, @iszmais! From my point of view, everything is fine. As the procedure is, some colleagues still need to look over it and give their approval. I hope they will do so soon.

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: When I read your version, I had problems understanding the part "shares of information."

Can you slightly change to: 
"The component only handles the contact and sharing of information via the contact functionality. There are multiple other types of context-related contacts in ILIAS that are not part of this component, such as course member/tutor relations or test correction processes, etc."

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: Please correct: "a user profiles" to "a user profile" or "user profiles".

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: Can you slightly change to:
"The component does not track or recognize the data being exchanged through emails or chats. It has no information about the source of the content (i.e. a course) or whether it includes user-related data."

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: Please add "." at the end of the sentence.

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: You can delete "by" here.

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: Can you slightly change to:
"Any user relations that originate from the contact are not related to or dependent on the active contact. There is no traceable relation of contacts within other components."

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: You can change "within" to "in".

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md]: Can you slightly change to:
"If a user is deleted, their contacts, sent and received contact requests, mailing lists, and mailing list participations are removed without any traceable footprint."

File: components/ILIAS/Contact/PRIVACY.md
Comment (selyesa) [inline components/ILIAS/Contact/PRIVACY.md:83]: Can you slightly change to:
A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that contains the user's presentation name (depending on their visibility settings) and the user's intention for the contact request.
The notification is handled by the "Notification" component, is self-persistent, and therefore not dependent on the origin request.

**Review by kseniialf — APPROVED** [review]

**Pull request merged**
